src/Subscriptions/StorageManager.php

```diff
@@ -110,8 +111,42 @@ public function storeSubscriber(Subscriber $subscriber, string $topic): void
         }
     }
 
+    /**
+     * Delete subscriber.
+     *
+     * @param  string  $channel
+     * @return \Nuwave\Lighthouse\Subscriptions\Subscriber|null
+     */
     public function deleteSubscriber(string $channel): ?Subscriber
     {
-        return $this->cache->pull(self::SUBSCRIBER_KEY.".{$channel}");
+        $key = self::SUBSCRIBER_KEY.".{$channel}";
+        $this->removeSubscriberFromTopic($key);
+
+        return $this->cache->pull($key);
```

Comment:
This still incurs an extra cache lookup, `->get()` is already called within `removeSubscriberFromTopic()`. We should keep IO to a minimum.